### PR TITLE
debugging: add common network errors

### PR DESCRIPTION
**Pull request by PettitWesley**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: troubleshooting/debugging.md [hunk 1]
```diff
@@ -98,6 +99,43 @@ Even if you see this message, you still have not lost logs yet. Since it will re
 
 When you see this message, you have lost logs. The other case that indicates log loss is when an input is paused, which is covered in the [overlimit error section](#overlimit-warnings).
 
+#### Common Network Errors
+
+First, please read the section [Network Connection Issues](#network-connection-issues), which explains that Fluent Bit's http client is very low level and logs errors that other clients may simply silently retry. That section also explains the `auto_retry_requests` option which now defaults to true.
```
Comment (ShelbyZ) [inline troubleshooting/debugging.md:104]: [nit-pick]
 - `very` in "very low level" does not add additional details
 - `auto_retry_requests` (defaults to true) - let's treat this as the present

File: troubleshooting/debugging.md [hunk 2]
```diff
@@ -98,6 +99,43 @@ Even if you see this message, you still have not lost logs yet. Since it will re
 
 When you see this message, you have lost logs. The other case that indicates log loss is when an input is paused, which is covered in the [overlimit error section](#overlimit-warnings).
 
+#### Common Network Errors
+
+First, please read the section [Network Connection Issues](#network-connection-issues), which explains that Fluent Bit's http client is very low level and logs errors that other clients may simply silently retry. That section also explains the `auto_retry_requests` option which now defaults to true. 
+
+Your Fluent Bit deployment will likely occasionally log network connection errors. These are normal and if they occur only occasionally and do not lead log loss, should not be a cause for concern. See [How do I tell if Fluent Bit is losing logs?](#how-do-i-tell-if-fluent-bit-is-losing-logs) to learn how to check for log loss from network issues. If you see network errors happen very frequently or they cause your retries to expire, then please cut us an issue.
```
Comment (ShelbyZ) [inline troubleshooting/debugging.md:106]: [nit-pick]
- likely occasionally -> occasionally
- These are normal and if they occur only occasionally and do not lead log loss, should not be a cause for concern - These are normal and should not cause concern unless...

**Review by ShelbyZ — APPROVED** [review]

File: troubleshooting/debugging.md [hunk 3]
```diff
@@ -395,6 +433,8 @@ In the AWS for Fluent Bit repo, we always attempt to tag bug reports with which
 
 One of the simplest causes of network connection issues is throttling, some AWS APIs will block new connections from the same IP for throttling (rather than wasting effort returning a throttling error in the response). We have seen this with the CloudWatch Logs API. So, the first thing to check when you experience network connection issues is your log ingestion/throughput rate and the limits for your destination. 
 
+Please see [Common Network Errors](#common-network-errors) for a list of common network errors outputtted by Fluent Bit.
```
Comment (matthewfala) [inline troubleshooting/debugging.md:436]: minor typo in outputtted

**Review by matthewfala — APPROVED** [review]
Found minor typo, but other then than that, looks good.

**Pull request merged**